livestream/views_test.py

```diff
@@ -0,0 +1,39 @@
+"""tests for the livestream views"""
+from django.urls import reverse
+from rest_framework import status
+
+
+def test_get_upcoming_events(client, user, mocker, settings):
+    """test that things work normally"""
+    settings.LIVESTREAM_ACCOUNT_ID = 3_234_234
+    settings.LIVESTREAM_SECRET_KEY = "secret ^_^"
+    client.force_login(user)
+    ls_api_mock = mocker.Mock()
+    ls_api_mock.configure_mock(
+        **{
+            "json.return_value": {
+                "total": 2,
+                "data": [{"first": "event"}, {"second": "event"}],
+            }
+        }
+    )
+    get_stub = mocker.patch(
+        "livestream.views.get_upcoming_events", return_value=ls_api_mock
+    )
+    resp = client.get(reverse("livestream"))
+    get_stub.assert_called_once()
+    assert resp.json() == {
+        "total": 2,
+        "data": [{"first": "event"}, {"second": "event"}],
+    }
+    assert resp.status_code == status.HTTP_200_OK
+
+
+def test_get_events_no_vars(client, user, settings):
```

Comment:
Can we parameterize this to also test if one or the other is missing? For example

    @pytest.mark.parametrize("has_account_id,has_secret_key", [
        [True, False],
        [False, True],
        [True, False],
    ])
    def test_get_events_no_vars(client, user, settings, has_account_id, has_secret_key):
        """we should get a 503 if the settings aren't present"""
        settings.LIVESTREAM_ACCOUNT_ID = "account_id" if has_account_id else None
        settings.LIVESTREAM_SECRET_KEY = "secret_key" if has_secret_key else None
        client.force_login(user)
        resp = client.get(reverse("livestream"))
        assert resp.json() == {}
        assert resp.status_code == status.HTTP_503_SERVICE_UNAVAILABLE